lib/ansible/galaxy/api.py

```diff
@@ -500,13 +500,20 @@ def get_collection_version_metadata(self, namespace, name, version):
         :param version: Optional version of the collection to get the information for.
         :return: CollectionVersionMetadata about the collection at the version requested.
         """
-        api_path = self.available_api_versions.get('v3', self.available_api_versions.get('v2'))
+        if 'v3' in self.available_api_versions:
+            api_path = self.available_api_versions['v3']
+            auth_required = True
```

Comment:
Just because it's `v3` doesn't mean auth is required. For Automation Hub, yes, `v3` requires auth to access metadata. In the future when community Galaxy has a `v3`, I suspect auth won't be required.

Reply:
Agreed. But at the moment, there is no other mechanism to determine if auth is required.